pkg/ephemeral/network/tls_connector_test.go

```diff
@@ -0,0 +1,228 @@
+	pemFileClient = `-----BEGIN CERTIFICATE-----
+MIIC+zCCAeOgAwIBAgIUWOrYZliAZd4NDKJBNkYsOqSCj5owDQYJKoZIhvcNAQEL
+BQAwDTELMAkGA1UEAwwCQzAwHhcNMjIwMTI2MTI0NjQ5WhcNMjIwMjI1MTI0NjQ5
+WjANMQswCQYDVQQDDAJDMDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEB
+AK/HZ55UNcblH6UaWyWdg0eUmCaaQt5efkxPXypNurcwKkHqjit3wjZrUP+4VSG3
+Xc66uPvOKUrP/laNxOehUbt+IItwGPe1e8IvnKY4bHGZB43myIlx5+kHyRxIHhGS
+s433LocKllIg4y+99JUytUytCP+uOqw7b9W/UD7icRMmbcaop/BKKSGA2xVyqIdg
+aSDR3X/fM/wJMBD6Wn3YuKN4kAwLI3Ieb8D5bJreDyF5FubqU+1lZRw85/KfcUZD
+4+70xsGtBw9tc9bSyO0agHB40thjKezfJWJKnuDI6dlZoDCdC0DVPs880aQ5inl6
+EjeADhuGEJbFuQ1lXLSpqOECAwEAAaNTMFEwHQYDVR0OBBYEFCXac7qi2TG+j/CQ
+fVyvM6W3JfONMB8GA1UdIwQYMBaAFCXac7qi2TG+j/CQfVyvM6W3JfONMA8GA1Ud
+EwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggEBAE0xk3rMO3xmpq1mwWGGQ/B2
+J9Xlqf5qwr63MNz6aIcKrlyk2+OLLaDm8RrF7wFNQ+uvMKKg6bLF7jW7MAX9WMO7
+giiT5ySjxddDT0cbSA3HcG3Ria9P6c02VZVt057M1FzXweR/FiJA1Tocn43lXrBT
+n2sAiRtO4sxbfhUdIJI1Vh7UUhyAJLe3lVcG/AMMmPG/IedguhMbdalm5/gEaIIc
+LjHyQLPWzHQTiUvj+AjpTmCN+3ZbBS/8r4g7XJ7/zvawXxi1Lk9fvSGWGkQLwHJ0
+DupEw8GWmc9H0cyY93qtEqKLQPvEDDdvhPoENcf/P6/BD1Z8lMmSMvZ+s6M7VfQ=
+-----END CERTIFICATE-----
+`
+
+	keyFileServer = `-----BEGIN PRIVATE KEY-----
+MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQCwdCqmEjiMVBPK
+m31IG+I+xwgX+EnEpnrBnlOa0WhFzaMrwqXpijgMA+dLNR8a1zpyhglWBsqm8dpN
+7tEV19piizOmxZtZee7h1Hdso/+4U106NqzX5HKwuqZVSOjVN29SFKq0sNricIX1
+HabE5LYyBQJtzMzxAZwclb+e7uGBfHJDsOk3hOhs3bkJyV3eRa0uHH2Bu4CPH6L9
+bcisFCmHiykZeZaY+BpRkS0c5+h7umLrKSGUe37/vf9UY9niLDUNolHePS/iQnmb
+Hv1l/mDl2LvNy5OSCSvOE6L0GMCUDnYmYf6F999LLdQgC7gcZCp3rujZ4MYUsqR2
+Nqp46LdbAgMBAAECggEBAJ6ViM8AiTn1RmRNImdwSAHLtwZz6ziFtsXUmacGlQRH
+MGLf6WTfCEgkKfd5op7o2Gqc9D8Qk4k+y8hG3jsXZ/owyRcVee0MnRjxbvOA4Q60
+PZFYGjdd5YXX+i2j/T3DOJU4ZcNHPzFLl9kX8Q37z5Nc1TYBXh8sJzW5kCIy5xEL
+XAKNcwGTZF1ml3jkWkFl3LukS3DP8fF1qDvD987YGuc9oVliYW1F0oKL9VGyS7nB
+BtQWslFdP8MbPXG1hjkFydCBiE4teqrFen6hvLdIQk7XJ88Q9UmBoOPJr6+gHuDf
+vk33nVGpBVQ1UHFPnDzZyQKtlDBVEUJ8XhqzEkm4asECgYEA5sTgxJt/nJCL9Lh1
+61jFbVD21SVFEv7IWIV6YjBxzJhzGVJa6ZhrOnRTrkTAkraJ1wUd9FyIdEsL/Nvy
+/z8hOAXbty1zXdpOo/BV0J6zRwJ0Cj8WVTeCUr5KGgw/pzbQdltJ+1J8jnAGbZjN
+Ri/QUdryqZTQz3rD8sDVDFvLojMCgYEAw78HQ+y/gL5Z/IJww0lUYHjqcm1G5taY
+3Ht6qRvkqdCmW8qC2wpKFKl9lCJfo+H1jidjhM5RTPFSlCxiWtxLAamMvfv0f3d6
+q5gPjcjak275bnmU1e0blkLEdeXQljRXH+oDmur95udzh0DrdTDJ/lqbf3uui8Uc
+VApAcSbR/jkCgYAWUT/zg55Jw+jlF9m/kuw08DmOz3Xoql8xwGbfjBPVV4D6F+7W
+3HiyRIG7Psbo6WJXOxV0hmZj6MYWBCdx6+cIhfiDtI+Nqgkk7Z8+97oaye/y9brx
+LtcZrXF5J2oYf8KVT6rN9WI6XDci7j4b5Y/d+rCxGcU/6317wo5YDaCZ5QKBgET3
+4qRxHwxKhUQt5XM5PAx9rgVBMXEV/Wf57b71v/yBMow27yIkHvPmwANYlSAV9kHu
+6OabFxQoFvN0K/ddlOPyDE/IHV5oB4W8HwbS1QiLWkEtf15cm5K21afAoFy79lKd
+TkXgNDOOKytlmVCCLzl6TT1+o4JFofSOZCQ6DFUpAoGAAQdaCjX5UCeWb5e/Vbiu
+SQL1RKIHkgm6gj1UjlQ981r6y+hVkBygtIr/eW0wSkFAkUrOdefHNVOQW18ESF06
+YqBL4gD7aEij9kGd0PrievimgcYYaBHOcO1RouQOURTMmWqjIPu1fyWDv+rFk+S5
+2uCuYndpzOgCiEhjDGCuSug=
+-----END PRIVATE KEY-----`
+
+	pemFileServer = `-----BEGIN CERTIFICATE-----
+MIIC+zCCAeOgAwIBAgIUNI9WRun2Y+ICmpzjYRpVcJ/BBE4wDQYJKoZIhvcNAQEL
```

Comment:
cert seems to have expired already. shouldn't this produce an error?
![grafik](https://user-images.githubusercontent.com/43000614/162132206-fcc8be3b-d998-4aaf-8f47-fca80286f3e6.png)

Reply:
btw. same for client cert

Reply:
The way this PR introduces the TLS connections is pretty insecure (we use `InsecureSkipVerify: true`).
We need to generally think more about how to address TLS certificates and their re-creation.
Opened issue #22 for this

What do you think should be part of this PR?
Would a new Certificate with a long duration (e.g. somewhere in the 2030s) work?

```go
	tlsConfig := &tls.Config{
		Certificates:       []tls.Certificate{cert},
		InsecureSkipVerify: true,
	}
```

Reply:
Yes, let's do it this way @kindlich. As [discussed on Discord](https://discord.com/channels/938353101147086870/960758450009292810/963050848114602034), the way to go is anyway to make TLS channels optional in upstream MP-SPDZ and have authenticated channels implemented via Istio mechanisms.

Reply:
Done.